### DISCUSSION TITLE
Sorting of connection listview without --showallservers

### DISCUSSION THREAD

**Discussion by sthenos** [jamulussoftware/jamulus#1272] [UI Related]
Currently when you start jamulus client with `--showallservers` option you can click the headers of the listview to sort by name, ping, location, etc.   But if you remove this option you cannot click the headers.   It would be very useful in some circumstances to be able to sort by name or sort by location too.  Can this be enabled please?

**Comment by dcorson-ticino-com** [comment]
I have looked at this and it will be quite complicated because of the special treatment of the ping times to reduce the jumping around of the list.  Because of this treatment it appears that the ping times remain steady even when they aren't, but it means that the native list sorting can not be used.  I have not yet had the light go on for how to do this best, but will keep it on my desk.

**Comment by gilgongo** [comment]
@dingodoppelt has a branch with this in and it works for me https://github.com/jamulussoftware/jamulus/discussions/1146 so your wish may come true. There has been some grounds to oppose sorting the list. New users who haven't fully understood ping times may conclude in error that Jamulus is rubbish if they are in Amsterdam and want to join a jam in Tokyo.

BTW not wishing to fork the thread,  but can you describe why you are using `--showallservers`?

**Reply by dcorson-ticino-com** [reply]
Note that the solution allowing native sorting enables, of course, the sort function for the other columns, but if sorting on the ping, which I am sure most people will continue to do, the list jumps around much more as the "special treatment" is not applied.  For the ping sort it is suboptimal.
I think the best solution would be a two column sort, ping and XXX.  But that is not supported natively by Qt and would be some work to do.
